test/include/seqan3/test/sandboxed_path.hpp

```diff
@@ -0,0 +1,416 @@
+     * This works the same as std::filesystem::path::concat
+     * and additionally checks the invariant.
+     *
+     * Basic exception guarantee
+     */
+    template <typename Source>
+    sandboxed_path & concat(Source const & source)
+    {
+        std::filesystem::path::concat(source);
+        normalize();
+        checkInvariant();
+        return *this;
+    }
+
+    /*!\brief Extends the path.
+     *
+     * This works the same as std::filesystem::path::concat
+     * and additionally checks the invariant.
+     *
+     * Basic exception guarantee
+     */
+    template <typename InputIt>
+    sandboxed_path & concat(InputIt first, InputIt last)
+    {
+        std::filesystem::path::concat(first, last);
+        normalize();
+        checkInvariant();
+        return *this;
+    }
+
+    /*!\brief Removes the filename.
+     *
+     * This works the same as std::filesystem::path::remove_filename
+     * and additionally checks the invariant.
+     *
+     * Basic exception guarantee
+     */
+    sandboxed_path & remove_filename()
+    {
+        std::filesystem::path::remove_filename();
+        normalize();
+        checkInvariant();
+        return *this;
+    }
+
+    /*!\brief Replaces the filename.
+     *
+     * This works the same as std::filesystem::path::replace_filename
+     * and additionally checks the invariant.
+     *
+     * Basic exception guarantee
+     */
+    sandboxed_path & replace_filename(std::filesystem::path const & p)
+    {
+        std::filesystem::path::replace_filename(p);
+        normalize();
+        checkInvariant();
+        return *this;
+    }
+
+    /*!\brief Replaces the extension.
+     *
+     * This works the same as std::filesystem::path::replace_extension
+     * and additionally checks the invariant.
+     *
+     * Basic exception guarantee
+     */
+    sandboxed_path & replace_extension(std::filesystem::path const & replacement = std::filesystem::path{})
+    {
+        std::filesystem::path::replace_extension(replacement);
+        normalize();
+        checkInvariant();
+        return *this;
+    }
+
+    /*!\brief Returns sandboxed path to the parent path.
+     *
+     * This works the same as std::filesystem::path::parent_path
+     * and additionally checks the invariant.
+     *
+     * Basic exception guarantee
+     */
+    sandboxed_path parent_path() const
+    {
+        auto parent_path = std::filesystem::path::parent_path();
+        return sandboxed_path{sandbox_path, parent_path};
+    }
+
+
+    void swap(sandboxed_path &) noexcept = delete; //!< Swap operator is not possible
+
+private:
+    void clear() = delete; //!< Unlikely to be usefull, not implemented.
+};
+
+/** Free sandboxed_path append operator.
+ *  This work the same as std::filesystem::operator/(std::filesystem::path&)
+ *  and additionally checks the invariant.
+ */
+template <typename Rhs>
+sandboxed_path operator/(sandboxed_path lhs, Rhs const& rhs) {
+    lhs /= rhs;
+    return lhs;
+}
```

Comment:
You should be able to remove this.

std::filesystem::operator/(std::filesystem::path&) works via ADL and ADL also looks for base classes.